src/targets/__tests__/maven.test.ts

```diff
@@ -0,0 +1,198 @@
+import { homedir } from 'os';
+import { join } from 'path';
+import { NoneArtifactProvider } from '../../artifact_providers/none';
+import { MavenTarget, targetOptions, targetSecrets } from '../maven';
+import { retrySpawnProcess } from '../../utils/async';
+import { withTempDir } from '../../utils/files';
+
+jest.mock('../../utils/files');
+
+jest.mock('fs', () => ({
+  ...jest.requireActual('fs'),
+  promises: {
+    writeFile: jest.fn(() => Promise.resolve()),
+  },
+}));
+
+jest.mock('../../utils/system', () => ({
+  ...jest.requireActual('../../utils/system'),
+  checkExecutableIsPresent: jest.fn(),
+  extractZipArchive: jest.fn(),
+}));
+
+jest.mock('../../utils/async');
+
+const DEFAULT_OPTION_VALUE = 'my_default_value';
+
+function setTargetSecretsInEnv(): void {
+  for (const option of targetSecrets) {
+    process.env[option] = DEFAULT_OPTION_VALUE;
+  }
+}
+
+function removeTargetSecretsFromEnv(): void {
+  for (const option of targetSecrets) {
+    delete process.env[option];
+  }
+}
+
+function getTargetOptions() {
+  return {
+    OSSRH_USERNAME: DEFAULT_OPTION_VALUE,
+    OSSRH_PASSWORD: DEFAULT_OPTION_VALUE,
+    gradleCliPath: DEFAULT_OPTION_VALUE,
+    mavenCliPath: DEFAULT_OPTION_VALUE,
+    mavenSettingsPath: DEFAULT_OPTION_VALUE,
+    mavenRepoId: DEFAULT_OPTION_VALUE,
+    mavenRepoUrl: DEFAULT_OPTION_VALUE,
+    android: {
+      distDirRegex: '/distDir/',
+      fileReplaceeRegex: '/replacee/',
+      fileReplacerStr: 'replacer',
+    },
+  };
+}
+
+function createMavenTarget(
+  targetOptions?: Record<string, unknown>
+): MavenTarget {
+  const finalOptions = targetOptions ? targetOptions : getTargetOptions();
+  const mergedOptions = {
+    name: 'maven',
```

Comment:
Why do we set the name here? Isn’t that a class property?

Reply:
It is not. It is part of the config schema (since you need to define it in `.craft.yml` anyway). We enforce it to be a static string via TypeScript at compile-time and then use it for logger name: https://github.com/getsentry/craft/blob/e84f693f23ad2de33ca13c099460e2b46127a799/src/targets/base.ts#L36

We cannot use `constructor.name` as the code gets minified. See #228 and #274 for more details.